### ISSUE TITLE
intel-m10-bmc-hwmon N6000 board power value in milli Watts instead of micro-Watts required by HWMON SYSFS

### ISSUE THREAD

**Issue by umairsiddiqui-digitek** [OPAE/linux-dfl#25]
According to 
https://www.kernel.org/doc/Documentation/hwmon/sysfs-interface

HWMon expects board power in microwatts units

```
power[1-*]_input		Instantaneous power use
				Unit: microWatt
				RO
				
```

However the BMC of N6001 provides board power value in milli Watts. therefore following line should be

https://github.com/OFS/linux-dfl/blob/6b0d820d436dda1e5949205a8f081fe0360c8430/drivers/hwmon/intel-m10-bmc-hwmon.c#L546

```
static const struct m10bmc_sdata n6000bmc_power_tbl[] = {
	{ 0x724, 0x0, 0x0, 0x0, 0x0, 1000, "Board Power" },
};
```

Now the 

```
fpgainfo bmc
```

shows the correct value, because OPAE SDK reads from sysfs and treats the value in milli Watts

https://github.com/OFS/opae-sdk/blob/31d4aba7e9b456f29234a97ce5c215f00d82e340/libraries/plugins/xfpga/metrics/metrics_max10.c#L387

But it is incorrect from hwmon sysfs prospective

**Comment by pcolberg** [comment]
Thanks for the report; I could confirm the behaviour on an N6001.

```
# strace -o fpgainfo_bmc_strace.log fpgainfo bmc | grep -i power
(45) Board Power                                        : 27.32 Watts

# grep -A2 hwmon.*power fpgainfo_bmc_strace.log
lstat("/sys/class/fpga_region/region0/dfl-fme.0/dfl_dev.6/n6000bmc-hwmon.2.auto/hwmon/hwmon4/power1_label", {st_mode=S_IFREG|0444, st_size=4096, ...}) = 0
openat(AT_FDCWD, "/sys/class/fpga_region/region0/dfl-fme.0/dfl_dev.6/n6000bmc-hwmon.2.auto/hwmon/hwmon4/power1_label", O_RDONLY) = 4
fstat(4, {st_mode=S_IFREG|0444, st_size=4096, ...}) = 0
read(4, "Board Power\n", 4096)          = 12
--
openat(AT_FDCWD, "/sys/class/fpga_region/region0/dfl-fme.0/dfl_dev.6/n6000bmc-hwmon.2.auto/hwmon/hwmon4/power1_input", O_RDONLY) = 4
lseek(4, 0, SEEK_SET)                   = 0
read(4, "27319\n", 256)                 = 6

# sensors | grep -i 'board power'
Board Power:                                   27.32 mW 
```

The issue was introduced with upstream commit e1983220ae14ae7bdc8db4fb64e78bb55122c32b.

**Comment by pcolberg** [comment]
I will submit a fix to the upstream kernel and to OPAE SDK. I briefly considered whether the fix for OPAE SDK should include a heuristic to work with both broken and fixed kernels, but that would likely make things worse when the hardware (BMC) is actually reporting wrong values, which might then be shadowed by the heuristic.

**Comment by pcolberg** [comment]
Submitted https://github.com/OFS/linux-dfl-backport/pull/118 to Linux DFL backport driver.

@umairsiddiqui-digitek, please let me know your name and email address if you would like to be acknowledged publicly in the kernel commit message.

**Comment by umairsiddiqui-digitek** [comment]
after this update opae sdk need to be updated as well

https://github.com/OFS/opae-sdk

because

`fpgainfo bmc `

assumes power in milli watts

https://github.com/OFS/opae-sdk/blob/31d4aba7e9b456f29234a97ce5c215f00d82e340/libraries/plugins/xfpga/metrics/metrics_max10.c#L387

**Comment by pcolberg** [comment]
Thanks @umairsiddiqui-digitek.

Submitted [hwmon: intel-m10-bmc-hwmon: fix multiplier for N6000 board power sensor](https://patchwork.kernel.org/project/linux-hwmon/patch/20240521181246.683833-1-peter.colberg@intel.com/) upstream.

**Comment by pcolberg** [comment]
Fixed in https://github.com/OFS/opae-sdk/commit/6ba858176fd11fe5e1680d34cf2b207a2d864224, https://github.com/OFS/linux-dfl-backport/commit/2fd7bfa558d257dd90a9f58355c5bc6326dcc2fd, and https://github.com/OFS/linux-dfl/commit/c2e016414a996cbf24f567ca14e80bb1dffd2913.

**Comment by pcolberg** [comment]
The kernel fix was upstreamed to Linux 6.10 as commit [027a44fedd55 ("hwmon: (intel-m10-bmc-hwmon) Fix multiplier for N6000 board power sensor")](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=027a44fedd55fbdf1d45603894634acd960ad04b).
